java/code/src/com/redhat/rhn/domain/scc/SCCCachingFactory.java

```diff
@@ -190,20 +184,24 @@ public static SCCSubscription lookupSubscriptionBySccId(Long id) {
         if (id == null) {
             return null;
         }
-        Session session = getSession();
-        Criteria c = session.createCriteria(SCCSubscription.class);
-        c.add(Restrictions.eq("sccId", id));
-        return (SCCSubscription) c.uniqueResult();
+        try {
+            return getSession().createNativeQuery("""
+                                      SELECT * from suseSCCSubscription
+                                      WHERE scc_id = :scc
+                                      """, SCCSubscription.class)
+                    .setParameter("scc", id , LongType.INSTANCE)
+                    .getSingleResult();
+        }
+        catch (NoResultException e) {
+            return null;
+        }
     }
+
     /**
      * Clear all subscriptions from the database.
      */
     public static void clearSubscriptions() {
-        CriteriaBuilder builder = getSession().getCriteriaBuilder();
-        CriteriaDelete<SCCSubscription> delete = builder.createCriteriaDelete(SCCSubscription.class);
-        delete.from(SCCSubscription.class);
-        getSession().createQuery(delete).executeUpdate();
-
+        getSession().createNativeQuery("DROP TABLE suseSCCSubscription CASCADE");
```

Comment:
This is wrong. We want to drop all data, but not the table.
```suggestion
        getSession().createNativeQuery("DELETE FROM suseSCCSubscription");
```